src/main/kotlin/org/openrewrite/RecipeMarkdownGenerator.kt

```diff
@@ -163,14 +170,11 @@ class RecipeMarkdownGenerator : Runnable {
 
             // Changes something like org.openrewrite.circleci.InstallOrb to https://docs.openrewrite.org/recipes/circleci/installorb
             val docLink = "https://docs.openrewrite.org/recipes/" + getRecipePath(recipeDescriptor)
-            val recipeSource = recipeDescriptor.source.toString()
-            var isImperative = true
 
-            // YAML recipes will have a source that ends with META-INF/rewrite/something.yml
+            // Determine if recipe is imperative (Java) or declarative (YAML)
             // Used to help with time spent calculations. Imperative = 12 hours, Declarative = 4 hours
-            if (recipeSource.substring(recipeSource.length - 3) == "yml") {
-                isImperative = false
-            }
+            val recipe = recipesByName[recipeDescriptor.name]
+            val isImperative = recipe !is DeclarativeRecipe
```

Comment:
This previously was marking some recipes as imperative when they weren't - because they were being defined in `.yaml` instead of `.yml` files.